apps/rewards/app/components/App/App.js

```diff
@@ -196,8 +197,12 @@ class App extends React.Component {
   }
 
   claimReward = reward => {
-    // TODO
-    this.props.api.claimReward(reward.rewardId + reward.claims).toPromise()
+    this.props.api.claimReward(reward.rewardId + reward.claims)
+      .subscribe(claimHash => {
+        const { claimHashes } = this.state
```

Comment:
Is it worth declaring a `const` here? I feel like it adds clarity to what I'm doing.

Reply:
I guess an alternative would be:

```js
this.setState({ claimHashes: {
  ...this.state.claimHashes,
  [reward.rewardId]: claimHash,
}})
```

I'm not sure how I feel about my formatting; hopefully eslint has opinions that we can defer to.

I don't think my suggestion or the way you wrote this read more clearly. I have questions either way!

* Do you think `claimedHashes` is a more accurate name? These are past-tense _claimed_, right? 
* `claimHashes` is initialized to an empty object, and it's only updated when a reward is claimed. If the user refreshes the page, this link will go away. Am I understanding correctly? Is that what we want?

Reply:
> `claimHashes` is initialized to an empty object, and it's only updated when a reward is claimed. If the user refreshes the page, this link will go away.
:scream:

Reply:
> Do you think `claimedHashes` is a more accurate name? These are past-tense claimed, right?

You're the English native speaker, but the hashes themselves aren't the thing being claimed. Instead, the hashes refer to claims. So the `claim` in `claimHashes` is a noun rather than a verb. Still, if it doesn't sound right to you, let's consider other options: `claimsHashes` seems more accurate to me because there are many claims, but it sounds bad. Any other ideas?

Reply:
> I don't think my suggestion or the way you wrote this read more clearly. I have questions either way!

To me, my way reads more clearly, although maybe I'm not being objective.

Reply:
I'm leaning toward "get rid of `claimHashes` altogether," as mentioned [here](https://github.com/AutarkLabs/open-enterprise/pull/1653#discussion_r354511112) and [here](https://github.com/AutarkLabs/open-enterprise/pull/1653#discussion_r354512239). We don't have to stick with the current structure of `rewards` that we have now; we can add `transactionHash` or, better yet, `etherscanUrl` to each reward, and simplify all the logic we have here.